src/xrpld/app/tx/detail/AMMWithdraw.cpp

```diff
@@ -552,6 +552,29 @@ AMMWithdraw::withdraw(
         return {tecAMM_BALANCE, STAmount{}};
     }
 
+    // Check the reserve in case a trustline has to be created
+    bool const enabledFixAMMv1_2 = view.rules().enabled(fixAMMv1_2);
+    auto sufficientReserve = [&](Issue const& issue) -> TER {
+        if (!enabledFixAMMv1_2 || isXRP(issue))
+            return tesSUCCESS;
+        if (!view.exists(keylet::line(account_, issue)))
+        {
+            auto const sleAccount = view.read(keylet::account(account_));
+            if (!sleAccount)
+                return tecINTERNAL;  // LCOV_EXCL_LINE
+            auto const balance = (*sleAccount)[sfBalance].xrp();
+            auto const reserve =
+                view.fees().accountReserve((*sleAccount)[sfOwnerCount] + 1);
```

Comment:
This should use the [same reserve logic as `SetTrust`](https://github.com/ximinez/rippled/blob/8d0b19cd6ed98e2e96d37e46254c111546761877/src/xrpld/app/tx/detail/SetTrust.cpp#L207-L229).
```suggestion
            std::uint32_t const ownerCount = sle->at(sfOwnerCount);

            // See also SetTrust::doApply()
            XRPAmount const reserve(
                (ownerCount < 2) ? XRPAmount(beast::zero)
                          : view().fees().accountReserve(ownerCount + 1));
```
It might be worth refactoring this into a helper function because the same logic is used in MPT, too.